cmd/bootstrap-peer-server.go

```diff
@@ -0,0 +1,248 @@
+	xhttp "github.com/minio/minio/cmd/http"
+	"github.com/minio/minio/cmd/logger"
+	"github.com/minio/minio/cmd/rest"
+)
+
+const (
+	bootstrapRESTVersion       = "v1"
+	bootstrapRESTVersionPrefix = SlashSeparator + bootstrapRESTVersion
+	bootstrapRESTPrefix        = minioReservedBucketPath + "/bootstrap"
+	bootstrapRESTPath          = bootstrapRESTPrefix + bootstrapRESTVersionPrefix
+)
+
+const (
+	bootstrapRESTMethodVerify = "/verify"
+)
+
+// To abstract a node over network.
+type bootstrapRESTServer struct{}
+
+// ServerSystemConfig - captures information about server configuration.
+type ServerSystemConfig struct {
+	MinioVersion  string
+	MinioPlatform string
+	MinioRuntime  string
+	MinioCmdLine  []string
+}
+
+// Diff - returns error on first difference found in two configs.
+func (s1 ServerSystemConfig) Diff(s2 ServerSystemConfig) error {
+	if s1.MinioVersion != s2.MinioVersion {
+		return fmt.Errorf("Expected server version %s, got %s", s1.MinioVersion,
+			s2.MinioVersion)
+	}
+	if !reflect.DeepEqual(s1.MinioCmdLine, s2.MinioCmdLine) {
+		return fmt.Errorf("Expected server command line %s, seen %s", s1.MinioCmdLine,
+			s2.MinioCmdLine)
+	}
+	if s1.MinioPlatform != s2.MinioPlatform {
+		return fmt.Errorf("Expected platform '%s', found to be running '%s'",
+			s1.MinioPlatform, s2.MinioPlatform)
+	}
+	if s1.MinioRuntime != s2.MinioRuntime {
+		return fmt.Errorf("Found servers running with different go-runtime versions, expected '%s', found '%s'",
+			s1.MinioRuntime, s2.MinioRuntime)
+	}
+	return nil
+}
+
+func getServerSystemCfg() ServerSystemConfig {
+	return ServerSystemConfig{
+		MinioVersion:  Version,
+		MinioPlatform: fmt.Sprintf("OS: %s | Arch: %s", runtime.GOOS, runtime.GOARCH),
+		MinioRuntime:  fmt.Sprintf("Version: %s | CPUs: %d", runtime.Version(), runtime.NumCPU()),
+		MinioCmdLine:  os.Args[1:],
+	}
+}
+
+func (b *bootstrapRESTServer) VerifyHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "VerifyHandler")
+	cfg := getServerSystemCfg()
+	logger.LogIf(ctx, gob.NewEncoder(w).Encode(&cfg))
+	w.(http.Flusher).Flush()
+}
+
+// registerBootstrapRESTHandlers - register bootstrap rest router.
+func registerBootstrapRESTHandlers(router *mux.Router) {
+	server := &bootstrapRESTServer{}
+	subrouter := router.PathPrefix(bootstrapRESTPrefix).Subrouter()
+
+	subrouter.Methods(http.MethodGet).Path(bootstrapRESTVersion + bootstrapRESTMethodVerify).HandlerFunc(
+		server.VerifyHandler)
+
+	// If none of the routes match add default error handler routes
+	router.NotFoundHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
+	router.MethodNotAllowedHandler = http.HandlerFunc(httpTraceAll(errorResponseHandler))
+}
+
+// client to talk to bootstrap Nodes.
+type bootstrapRESTClient struct {
+	endpoint   Endpoint
+	restClient *rest.Client
+	connected  int32
+}
+
+// Reconnect to a bootstrap rest server.
+func (client *bootstrapRESTClient) reConnect() {
+	atomic.StoreInt32(&client.connected, 1)
```

Comment:
Why are we changing the status right away? Shouldn't we actually try to reconnect?

Reply:
reConnect() is just a boolean to indicate the call to go through.